book/bayesian_optimization.qmd

```diff
@@ -0,0 +1,844 @@
+  ylim(c(0, 1))
+g_lhs = ggMarginal(g_lhs, type = "histogram", bins = 10)
+
+g_sobol = ggplot(aes(x = x1, y = x2), data = sobol_design) +
+  geom_point(size = 3L) +
+  geom_vline(xintercept = qs, linetype = 2) +
+  geom_hline(yintercept = qs, linetype = 2) +
+  labs(title = "Sobol Design", x = expression(x[1]), y = expression(x[2])) +
+  theme_minimal() +
+  xlim(c(0, 1)) +
+  ylim(c(0, 1))
+g_sobol = ggMarginal(g_sobol, type = "histogram", bins = 10)
+
+grid.arrange(g_random, g_grid, g_lhs, g_sobol, nrow = 2, ncol = 2)
+```
+
+We observe that a random design does not cover the domain well and simply by chance samples points close to each other.
+A grid design will always result in points being marginally equidistant from their nearest neighbour and does not cover the domain well.
+In contrast, an LHS design will result in good coverage of the domain and even for a small number of samples the marginal distributions will be perfectly uniform (i.e., in the plot, in each row and column there is exactly one sample).
+A Sobol design has a similar goal in mind but must not guarantee this for a small number of samples.
+However, constructing a Sobol design can be done much more efficiently than an LHS design, especially if the number of samples and dimensions grows.
+Moreover, a Sobol design has better coverage properties than an LHS design if the number of dimensions grows large.
+
+Of course, one can always evaluate a custom initial design in the form of a `data.table`.
+Coming back to our running example, we will now evaluate the following custom design:
+
+```{r}
+instance = OptimInstanceSingleCrit$new(objective,
+  terminator = trm("evals", n_evals = 20))
+design = data.table(x = c(0.1, 0.3, 0.65, 1))
+instance$eval_batch(design)
+```
+
+### Loop Function
+
+The `r ref("loop_function")` determines the behavior of the BO algorithm on a global level, i.e., how the subroutine should look like that is performed at each iteration.
+
+To get an overview of readily available loop functions, the following dictionary can be inspected:
+
+```{r}
+as.data.table(mlr_loop_functions)
+```
+
+Technically, all `loop_function`s are members of the `S3` class `r ref("loop_function")` (and are actually simply `r ref("base::function", "functions")` with some attributes set).
+
+For sequential single-objective black box optimization, the Efficient Global Optimization (EGO) algorithm [@jones_1998] can be considered the reference algorithm.
+In `r mlr3mbo`, the EGO algorithm is implemented via the `r ref("mlr_loop_functions_ego", "bayesopt_ego")` loop function.
+After having made some assertions and safety checks, and having evaluated the initial design, `bayesopt_ego` essentially repeatedly performs the following steps:
+
+1. `acq_function$surrogate$update()`: update the surrogate model
+2. `acq_function$update()`: update the acquisition function
+3. `acq_optimizer$optimize()`: optimize the acquisition function to yield a new candidate
+
+### Surrogate
```

Comment:
Whole section a bit hard to follow, don't have a good suggestion right now for how best to improve but basically there's just a lot of new and technical content